src/objects.c

```diff
@@ -32,6 +32,7 @@ struct p11prov_crt {
 struct p11prov_obj {
     P11PROV_CTX *ctx;
     bool raf; /* re-init after fork */
+    bool dup; /* duplicated */
 
```

Comment:
I do not think we need a new boolean here.
We have the poolid integer, if the object has not been added to a pooolid you can simply set poollid to a special value to indicate that. That value can be MAX(int) for example.